mesa/visualization/templates/js/GridDraw.js

```diff
@@ -196,89 +228,96 @@ var GridVisualization = function(width, height, gridWidth, gridHeight, context)
                         p4_y = cy + arrowR;
                 }
 
-		context.beginPath();
+                context.beginPath();
                 context.moveTo(p1_x, p1_y);
                 context.lineTo(p2_x, p2_y);
                 context.lineTo(p3_x, p3_y);
                 context.lineTo(p4_x, p4_y);
-		context.closePath();
+                context.closePath();
+
+                context.strokeStyle = stroke_color;
+                context.stroke();
 
-		context.strokeStyle = color;
-		context.stroke();
+                if (fill) {
+                        var gradient = context.createLinearGradient(p1_x, p1_y, p3_x, p3_y);
 
-		if (fill) {
-			context.fillStyle = color;
-			context.fill();
-		}
+                        for (i = 0; i < colors.length; i++) {
+                                gradient.addColorStop(i/colors.length, colors[i]);
+                        }
+
+                        // Fill with gradient
+                        context.fillStyle = gradient;
+                        context.fill();
+                }
 
                 // This part draws the text inside the ArrowHead
                 if (text !== undefined) {
-        		var cx = (x + 0.5) * cellWidth;
-                  	var cy = (y + 0.5) * cellHeight;
+                        var cx = (x + 0.5) * cellWidth;
+                          var cy = (y + 0.5) * cellHeight;
                         context.fillStyle = text_color
                         context.textAlign = 'center';
                         context.textBaseline= 'middle';
                         context.fillText(text, cx, cy);
                 }
         };
 
-	this.drawCustomImage = function (shape, x, y, scale, text, text_color_) {
-		var img = new Image();
-			img.src = "local/".concat(shape);
-		if (scale === undefined) {
-			var scale = 1
-		}
-		// Calculate coordinates so the image is always centered
-		var dWidth = cellWidth * scale;
-		var dHeight = cellHeight * scale;
-		var cx = x * cellWidth + cellWidth / 2 - dWidth / 2;
-		var cy = y * cellHeight + cellHeight / 2 - dHeight / 2;
-
-		// Coordinates for the text
-		var tx = (x + 0.5) * cellWidth;
-		var ty = (y + 0.5) * cellHeight;
-
-
-		img.onload = function() {
-			context.drawImage(img, cx, cy, dWidth, dHeight);
-			// This part draws the text on the image
-			if (text !== undefined) {
-				// ToDo: Fix fillStyle
-				// context.fillStyle = text_color;
-				context.textAlign = 'center';
-				context.textBaseline= 'middle';
-				context.fillText(text, tx, ty);
-			}
-		}
-	}
-
-	/**
+        this.drawCustomImage = function (shape, x, y, scale, text, text_color_) {
```

Comment:
I don't see any changes in this block. Tabs for spaces?

Reply:
Yes, tabs vs spaces.